Mathlib/AlgebraicGeometry/PrimeSpectrum/Basic.lean

```diff
@@ -636,6 +629,66 @@ lemma exists_idempotent_basicOpen_eq_of_is_clopen {s : Set (PrimeSpectrum R)}
     exact PrimeSpectrum.zeroLocus_anti_mono
       (Set.singleton_subset_iff.mpr <| Ideal.pow_le_self hnz hx)
 
+section IsIntegral
+
+open Polynomial
+
+theorem comap_singleton_isClosed_of_isIntegral {R S : Type*} [CommRing R] [CommRing S]
+    (f : R →+* S) (hf : f.IsIntegral)
+    (x : PrimeSpectrum S) (hx : IsClosed ({x} : Set (PrimeSpectrum S))) :
+    IsClosed ({comap f x} : Set (PrimeSpectrum R)) :=
+  have := (isClosed_singleton_iff_isMaximal x).1 hx
+  (isClosed_singleton_iff_isMaximal _).2
+    (Ideal.isMaximal_comap_of_isIntegral_of_isMaximal' f hf x.asIdeal)
+
+lemma closure_image_comap_zeroLocus {R S : Type*} [CommRing R] [CommRing S]
+    (f : R →+* S) (I : Ideal S) :
+    closure (comap f '' zeroLocus I) = zeroLocus (I.comap f) := by
+  apply subset_antisymm
+  · rw [(isClosed_zeroLocus _).closure_subset_iff, Set.image_subset_iff, preimage_comap_zeroLocus]
+    exact zeroLocus_anti_mono (Set.image_preimage_subset _ _)
+  · rintro x (hx : I.comap f ≤ x.asIdeal)
+    obtain ⟨q, hq₁, hq₂⟩ := Ideal.exists_minimalPrimes_le hx
+    obtain ⟨p', hp', hp'', rfl⟩ := Ideal.exists_comap_eq_of_mem_minimalPrimes f _ hq₁
+    let p'' : PrimeSpectrum S := ⟨p', hp'⟩
+    apply isClosed_closure.stableUnderSpecialization ((le_iff_specializes
+      (comap f ⟨p', hp'⟩) x).mp hq₂) (subset_closure (by exact ⟨_, hp'', rfl⟩))
+
+lemma isIntegral_of_isClosedMap_comap_polynomial
```

Comment:
```suggestion
lemma _root_.Algebra.isIntegral_of_isClosedMap_comap_polynomial
```
maybe? (to make it more discoverable)

Reply:
I think it has comap in its name so it should live in the `PrimeSpectrum` namespace.

Reply:
Hmm, so maybe `Algebra.isIntegral_of_isClosedMap_primeSpectrum_comap_polynomial`? Both are fine, I leave it up to you.